src/python/pants/jvm/test/junit_test.py

```diff
@@ -606,6 +612,178 @@ class SimpleTest {
     assert re.search(r"1 tests found", test_result.stdout) is not None
 
 
+@maybe_skip_jdk_test
+def test_vintage_file_dependency(rule_runner: RuleRunner) -> None:
```

Comment:
It would be good to extract a test helper across these three methods... maybe something that takes the `BUILD` file content and the final filename to expect as arguments...?